src/lib/hooks/useInfiniteScroll.ts

```diff
@@ -0,0 +1,26 @@
+import { useEffect, useCallback } from 'react';
+
+const useInfiniteScroll = (
+  loadMore: () => void,
+  hasMore: boolean | undefined
+) => {
+  const handleScroll = useCallback(() => {
+    const isAtBottom =
+      window.innerHeight + document.documentElement.scrollTop >=
+      document.documentElement.offsetHeight - 100;
+
+    // hasMore, 추가 데이터를 요청할 상태
+    if (isAtBottom && hasMore) {
+      loadMore();
+    }
+  }, [loadMore, hasMore]);
+
+  useEffect(() => {
+    window.addEventListener('scroll', handleScroll);
+    return () => {
+      window.removeEventListener('scroll', handleScroll);
+    };
+  }, [handleScroll]);
+};
```

Comment:
```suggestion
  const targetRef = useRef(null);
  // 스크롤맨 아래에 요소를 추가하고 ref를 참조하게 만듭니다

  useEffect(() => {
    const observer = new IntersectionObserver(
      (entries) => {
        //해당 요소가 보일때 실행 할 함수 
        }); 
        if(hasmore) loadMore()
      },
      { threshold: 1.0 }
    );

    if (!target) return;
    observer.observe(target);

    return () => {
      if (observer) {
        observer.unobserve(target);
      }
    };
  }, [target]);

```
p5: intersectionObserver Api를 사용해 스크롤 이벤트 없이 무한 스크롤을 구현할 수 있습니다!

Reply:
맞습니다. intersectionObserver를 사용하면 불필요한 이벤트 요청을 줄일 수 있어서 굉장히 효율적인 방식입니다. 채택하길 적극 권장합니다!!